### PR TITLE
Integrated Infiltration in autopilot

### PR DESCRIPTION
**Pull request by otto-kirchheim**
I built a service for "infiltration" and integrated it into "autopilot.js".
By the way, I removed the bugfix in "work-for-factions.js" for the update to Bitburner v2.1.0.

### CODE REVIEW

**Comment by alainbryden** [comment]
There's a lot of great work here, and I want to leave the PR up for those who want to see and try your branch, but I'm also seeing a few tangential changes mixed in (like disabling casino, and work-for-factions from focusing) which could be cause for debate. I'll test and consider merging changes piecemeal.

**Comment by cyberze** [comment]
I believe this to be the best way to leave some comments to that part of the script so:
- the fact that casino is disabled is good since you can earn money way faster with infiltration unless we are in SF8 (which should be auto implemented into autopilot)
- if running autopilot with the infiltration enabled, it should really run as much as possible, so after getting rep when apropriate, and nothing else is beeing done, it should run for more money, which is not the case
- while the infiltrator is running, it should not be stopped by other functions, as it leads to a lot of lost profit
- there is zero point in having both casino and infiltrator disabled on default, one should be always enabled, preferably first casino as it always works, and then switch to infiltrator as a calback from casino
- the posibility of earning with infiltration should be taken into account when autopilot decides time to ascend

**Comment by alainbryden** [comment]
All good comments.

Note that the reason I've been hesitant to include infiltration automation in my scripts is because (like Casino) there is no built-in API for it, it wasn't designed/balanced to be automated. As such, those who do manage to automate it end up with an overpowered mechanic that dominates the game in (almost) every BN.

Corporations have the same problem. Due to balance issues, and easy access to the investor fraud exploit, they are overpowered and a fun-ruiner. I'm sure an extensive debate could be had about how I've already ruined the fun by automating the majority of the game in my scripts. It's a delicate line to draw, but this is where I've decided to draw it.

I let casino.js exist because the game includes a mechanic that acknowledges it's exploitabolity and caps earnings. There's no cap (yet) on automated infiltration earnings, or exploited corporation earnings, so I'm steering clear of, them lest they become the **only** mechanic of any significance.

**Comment by cyberze** [comment]
All good points, but imho the soft cap with infiltration is that earning never change, hence after a few runs it becomes less viable than earnings from other ventures.

**Comment by teunisv** [comment]
Infiltration money/reputation exp lowers based on chance (str/def/dex/agi/cha) and SoA aurguments 
(for best rewards have both as low as possible)

I would love to see this implanted though

**Comment by teunisv** [comment]
Sorry semi hi-jack this post. Once I got fully working I’ll make my own pull request.
 
But is there a way to get Neuroflux reputations requirements, so that I can have infiltration focus on reputation before it focuses on money ?

**Comment by Ricochetv1** [comment]
Small thing. Noticed the guard attack routine is not working correctly. It should be attacking when it says "Preparing?" to succeed. If it gets to "ATTACKING!" it's too late. Also, I have the augment where it shows the countdown bar until the guard is ready to be attacked, and I'm not sure if that is a factor when it doesn't work.

File: infiltrate.js [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+      if (!state.game.data || !state.game.data.length) {
+        delete state.game.data;
+        return;
+      }
+
+      pressKey(state.game.data.shift());
+    },
+  },
+  {
+    name: "slash when his guard is down",
```
Comment (adomixaszvers) [inline infiltrate.js:111]: `slash` was replaced with `attack` on https://github.com/bitburner-official/bitburner-src/commit/f943e5d603cad3b42a0923508f0417abd44560d8

File: infiltrate.js [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+      const data = getLines(getEl(screen, "h4"));
+
+      if ("attack" === state.game.data) {
+        pressKey(" ");
+        state.game.data = "done";
+      }
+
+      // Attack in next frame - instant attack sometimes
+      // ends in failure.
+      if ("wait" === state.game.data && -1 !== data.indexOf("ATTACKING!")) {
```
Comment (adomixaszvers) [inline infiltrate.js:125]: The button should be pressed when the message is `Preparing?`. See https://github.com/bitburner-official/bitburner-src/blob/dev/src/Infiltration/ui/SlashGame.tsx#L36 and https://github.com/bitburner-official/bitburner-src/blob/dev/src/Infiltration/ui/SlashGame.tsx#L76

**Pull request closed without merging**